src/mlpack/methods/ann/activation_functions/mish_function.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file mish_function.hpp
+ * @author Kartik Dutt
+ *
+ * Definition and implementation of the mish function as described by
+ * Diganta Misra.
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @misc{
+ *   author = {Diganta Misra},
+ *   title = {Mish: Self Regularized Non-Monotonic Neural Activation Function},
+ *   year = {2019}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_MISH_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_MISH_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <algorithm>
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+
+namespace mlpack
+{
+namespace ann /** Artificial Neural Network. */
+{
+
+/**
+ * The mish function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = x * tanh(ln(1+e^x))
+ * f'(x) = tanh(ln(1+e^x)) + x * ((1 - tanh^2(ln(1+e^x))) * frac{1}{1 + e^{-x}})
+ * }
+ */
+class MishFunction
```

Comment:
Do you like to add an alias to the base layer e.g. https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/base_layer.hpp#L129

Reply:
Hi @zoq, Sure, would it be okay if I add swish here too (I have added it for now, I will remove it if needed)?

Reply:
Nice, catch, yes, good idea.